instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/package-info.java

```diff
@@ -0,0 +1,5 @@
+@DefaultQualifier(value = NonNull.class)
```

Comment:
This is still one point where all three repos aren't consistent and is on my list of things to address eventually. SDK uses `javax.annotation.Nullable`, which is less powerful than `checkerframework` since it doesn't support type annotations. Even if we don't use type annotations proactively, I'd like to converge on checkerframework version to at least have the ability for it without any real downside otherwise.

For the DefaultQualifier, we probably want to define our own meta-annotation that is something like

https://github.com/open-telemetry/opentelemetry-java-contrib/blob/8944db8459a51a5cb0500ecaf6f2903ec7546982/aws-xray/src/main/java/io/opentelemetry/contrib/awsxray/package-info.java#L7

Marking everything as NonNull by default is too much I've found (in particular, wildcard bounds become generally wrong).

Reply:
👍 added those same locations restrictions